mlir_graphblas/src/lib/GraphBLAS/GraphBLASOps.cpp

```diff
@@ -297,22 +297,26 @@ static LogicalResult verifyMatrixMultiplyArgs(T op)
 
   ArrayRef<int64_t> aShape = aTensorType.getShape();
   ArrayRef<int64_t> bShape = bTensorType.getShape();
-
-  // Vector-vector result is a scalar; Otherwise, get the tensor properties of the result
   int64_t resultRank = 0;
   ArrayRef<int64_t> resultShape;
   Type resultElementType = resultType;
-  if (aRank == 2 || bRank == 2) {
-    resultRank = getRank(resultType);
-    RankedTensorType resultTensorType = resultType.dyn_cast<RankedTensorType>();
-    resultShape = resultTensorType.getShape();
-    resultElementType = resultTensorType.getElementType();
+  RankedTensorType resultTensorType;
+
+  // Vector-vector result is a scalar; Otherwise, get the tensor properties of the result
+  if (checkResultTensorType)
+  {
+    if (aRank == 2 || bRank == 2)
+    {
+      resultTensorType = resultType.dyn_cast<RankedTensorType>();
+      resultShape = resultTensorType.getShape();
+      resultRank = getRank(resultType);
+      RankedTensorType resultTensorType = resultType.dyn_cast<RankedTensorType>();
```

Comment:
resultTensorType is duplicated in this line

Reply:
fixed